In [2]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [3]:
data=pd.read_csv('E:\Excelr files\data\Hypothesis testing\Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

chi2-test
Null hypothesis(Ho)=The defective % does not varies by centre.
Alternate hypothesis(H1)=The defective % does varies by centre.

In [7]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [8]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [9]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [10]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [11]:
data1=np.array([[271,267,269,280],[29,33,31,20]])
data1

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [12]:
chi2_contingency(data1) 

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

P value of Chi-Square test == 0.277 > α.
Null hypothesis accepeted.
The defective % does not varies by centre.